### PR TITLE
Improvement of several error messages

### PR DESCRIPTION
**Pull request by giorgiosironi**
This came out of trying to remaster elife-libraries-ci from the old builder; I discarded the rest of the modifications but the error messages were useful.

### CODE REVIEW

File: src/buildercore/bootstrap.py [hunk 1]
```diff
@@ -111,6 +111,8 @@ def stack_resources(stackname):
 
 def ec2_instance_data(stackname):
     "returns the ec2 instance data from the first ec2 instance the stack has"
+    if stackname is None:
```
Comment (lsh-0) [inline src/buildercore/bootstrap.py]: I prefer assertions to  raising RuntimeErrors. so, make these `assert stackname, "no stackname provided"`.

the absence of a stackname is programmer error and not something outside of our control.

File: src/master.py [hunk 2]
```diff
@@ -81,8 +81,8 @@ def remaster_minion(stackname):
     print 're-mastering',stackname
     download_keypair(stackname)    
     with core.stack_conn(stackname, username=config.BOOTSTRAP_USER):
-        sudo("rm /etc/salt/pki/minion/minion_master.pub")  # destroy the old master key we have
-    bootstrap.update_stack(stackname)    
+        sudo("rm -f /etc/salt/pki/minion/minion_master.pub")  # destroy the old master key we have
```
Comment (lsh-0) [inline src/master.py]: this is good actually, I encountered this yesterday with Ian

File: src/buildercore/bootstrap.py [hunk 3]
```diff
@@ -119,6 +121,8 @@ def ec2_instance_data(stackname):
 def master_data(region):
     "returns the ec2 instance data for the master-server"
     stackname = core.find_master(region)
+    if stackname is None:
```
Comment (lsh-0) [inline src/buildercore/bootstrap.py]: same again re assertions rather than runtime errors

File: src/aws.py [hunk 4]
```diff
@@ -33,7 +33,7 @@ def find_region(stackname=None):
     if len(region_list) > 1:
         if True:
             print "many possible regions found!"
-            return utils._pick(region, region_list)        
+            return utils._pick('region', region_list)
```
Comment (lsh-0) [inline src/aws.py]: how on earth did this make it past the linter? /rhetorical

**Comment by lsh-0** [comment]
looking good. if the conflicts could be resolved I'm happy to merge

**Comment by giorgiosironi** [comment]
Rebased to solve conflicts.

**Pull request merged**